src/player.cpp

```diff
@@ -139,6 +139,12 @@ HudElement* Player::getHud(u32 id)
 	return NULL;
 }
 
+void Player::hudApply(std::function<void(const std::vector<HudElement*>&)> f)
+{
+	MutexAutoLock lock(m_mutex);
+	f(hud);
```

Comment:
This mutex locking issue could be solved without a callback function: https://gist.github.com/SmallJoker/f75cc0e029e1ee989d59e88eab1138cc

I do not know whether other devs like it, thus I am leaving it here as an idea for feedback.

Reply:
Imo, the callback is fine.
Carrying a PtrLock everywhere is also not so nice.
Alternatively the mutex could be made available outside of the Player class, but this would just scatter things.

However, if anyone has a better idea that is supported by the core devs please make a suggestion.